app/models/creatorship.rb

```diff
@@ -4,6 +4,19 @@ class Creatorship < ActiveRecord::Base
 
   before_destroy :expire_caches
 
+  validate :unique_index
```

Comment:
Took me a moment to remember: I used it to make debugging easier when I was trying to figure out why creatorships were being created twice in some callbacks. This had to do with the order of the callbacks being defined that referenced associated models through join models (hence why it's in the same commit that moves the `import Taggable` line). Instead of an error message indicating that something had failed to save, I wanted an error message that told me explicitly why, and writing this is how I found it.

Since it's valid and indicates what's required explicitly I never removed it.